src/app_agent_websocket.rs

```diff
@@ -35,73 +49,72 @@ impl AppAgentWebsocket {
             my_pub_key: app_info.agent_pub_key.clone(),
             app_ws,
             app_info,
-            lair_client,
+            signer,
         })
     }
 
-    pub async fn on_signal<F: Fn(Signal) -> () + 'static + Sync + Send>(
+    pub async fn on_signal<F: Fn(Signal) + 'static + Sync + Send>(
         &mut self,
         handler: F,
     ) -> Result<String> {
         let app_info = self.app_info.clone();
         self.app_ws
-            .on_signal(move |signal| match signal.clone() {
-                Signal::App {
+            .on_signal(move |signal| {
+                if let Signal::App {
                     cell_id,
                     zome_name: _,
                     signal: _,
-                } => {
-                    if app_info
-                        .cell_info
-                        .values()
-                        .find(|cells| {
-                            cells
-                                .iter()
-                                .find(|cell_info| match cell_info {
-                                    CellInfo::Provisioned(cell) => cell.cell_id.eq(&cell_id),
-                                    CellInfo::Cloned(cell) => cell.cell_id.eq(&cell_id),
-                                    _ => false,
-                                })
-                                .is_some()
+                } = signal.clone()
+                {
+                    if app_info.cell_info.values().any(|cells| {
+                        cells.iter().any(|cell_info| match cell_info {
+                            CellInfo::Provisioned(cell) => cell.cell_id.eq(&cell_id),
+                            CellInfo::Cloned(cell) => cell.cell_id.eq(&cell_id),
+                            _ => false,
                         })
-                        .is_some()
-                    {
+                    }) {
                         handler(signal);
                     }
                 }
-                _ => {}
             })
             .await
     }
 
     pub async fn call_zome(
         &mut self,
-        role_name: RoleName,
+        target: ZomeCallTarget,
         zome_name: ZomeName,
         fn_name: FunctionName,
         payload: ExternIO,
     ) -> ConductorApiResult<ExternIO> {
-        let cell_id = self.get_cell_id_from_role_name(&role_name)?;
-
-        let agent_pub_key = self.app_info.agent_pub_key.clone();
+        let cell_id = match target {
```

Comment:
This whole thing is a bit of a mess. It's nice to have options when you call the zome but we don't refresh the `app_info` when we create/delete clone cells so looking at it to find `CellId`s is an obvious footgun. I've left some doc comments to help out a bit with this but with the `From` impls it's likely people won't even see those. This needs some more work!

Reply:
I see. I would like to either add a method to refresh the app info after clone cell operations or even remove all cell targets except `CellId` as that's the only safe one for now.

Reply:
Good call, I've added a refresh and updated the docs on the enum variants